### DISCUSSION TITLE
Enc/Dec Use Cases and Architectural choices

### DISCUSSION THREAD

**Discussion by bumblefudge** [ChainAgnostic/CAIPs#156] [Ideas]
Working in the open to solidify scope of the CAIP for enc/dec APIs

**Comment by coryhardman** [comment]
One general bit of advice is that with wallets we have really strong identity proofs. So rather than attempting to reuse the key material for enc/dec we should generate session keys for enc/dec situations using protocols like diffie-hellman key exchanges. 

Potentially some use cases to consider:

* Peer to peer messaging or data transfer using wallet identities
* Broadcast messaging to select identities 1 to many or many to many

The key thing is that your wallet acts as your identity in key establishment.

One great feature we get if we go a key establishment protocol: if you lose control of your wallet your previous messages are not compromised. Aka: [forward secrecy](https://en.wikipedia.org/wiki/Forward_secrecy)
